internal/controller/service/helpers.go

```diff
@@ -74,7 +82,13 @@ func (re *ReconcilerExtended) buildCNP() *ciliumv2.CiliumNetworkPolicy {
 	// }
 	// endpointSelectorLabelsMap := CNPlabels.StringMap()
 	// ##########################################################################
-	for k, v := range re.service.Spec.Selector {
+
+	identityLabels, informationLabels := cilium_labelsfilter.Filter(cilium_labels.Map2Labels(re.service.Spec.Selector, cilium_labels.LabelSourceK8s))
+	if len(informationLabels.K8sStringMap()) != 0 {
+		re.logger.Info("excluded labels", "information labels", informationLabels.String())
+	}
+
+	for k, v := range identityLabels.K8sStringMap() {
```

Comment:
We may instead use:
```Go
identityLabelSelector := v1.LabelSelector{MatchLabels: identityLabels.K8sStringMap()}
endpointSelector := api.NewESFromK8sLabelSelector(cilium_labels.LabelSourceK8sKeyPrefix, &identityLabelSelector)
```
and pass the `endpointSelector` directly to `EndpointSelector:` below.

Reply:
Thanks.